_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,442 @@
+  Normal   ExternalProvisioning  103s (x43 over 11m)  persistentvolume-controller                                                waiting for a volume to be created, either by external provisioner "dobs.csi.digitalocean.com" or manually created by system administrator
+```
+
+As instructed by the event message, I contacted DigitalOcean support and they fixed it.
+
+Finally, I had to open a second support ticket requesting higher node sizes, so I could use the size `s-4vcpu-8gb-intel`. Not all the size options where available for my account when I signed up.
+
+### Find your gateway's external IP and update redirect URIs
+
+Once all the pods are running, find the gateway external IP with the `kubectl describe` command:
+
+```bash
+kubectl describe service gateway -n demo
+```
+The output will look like:
+
+```text
+Name:                     gateway
+Namespace:                demo
+Labels:                   app=gateway
+Annotations:              kubernetes.digitalocean.com/load-balancer-id: e81d2b8c-6c28-430d-8ba8-6dab29a1ba76
+                          service.beta.kubernetes.io/do-loadbalancer-certificate-id: bd0b1d03-0f90-449d-abbe-ac6a4026c133
+Selector:                 app=gateway
+Type:                     LoadBalancer
+IP Family Policy:         SingleStack
+IP Families:              IPv4
+IP:                       10.245.47.51
+IPs:                      10.245.47.51
+LoadBalancer Ingress:     157.230.200.181
+Port:                     http  8080/TCP
+TargetPort:               8080/TCP
+NodePort:                 http  31048/TCP
+Endpoints:                10.244.1.42:8080
+Session Affinity:         None
+External Traffic Policy:  Cluster
+```
+
+Update the redirect URIs in Okta to allow the gateway address as a valid redirect. Run `okta login`, open the returned URL in your browser, and sign in to the Okta Admin Console. Go to the **Applications** section, find your application, edit and add:
+- Sign-in redirect URIs: `http://<load-balancer-ingress-ip>:8080/login/oauth2/code/oidc`
+- Sign-out redirect URIs: `http://<load-balancer-ingress-ip>:8080`
+
+Navigate to `http://<load-balancer-ingress-ip>:8080`.
+
+## Secure web traffic with HTTPS
+
+As the gateway service acts as the application front-end, in the k8s descriptors it is defined as a `LoadBalancer` service type. This exposes the service externally using the cloud provider's load balancer. DigitalOcean load balancers are a fully-managed, highly available network load balancing service. Load balancers distribute traffic to groups of droplets, which decouples the overall health of a backend service from the health of a single server to ensure that your services stay online.
```

Comment:
```suggestion
Since the gateway service acts as the application front-end, in the k8s descriptors it is defined as a `LoadBalancer` service type. This exposes the service externally using the cloud provider's load balancer. DigitalOcean load balancers are a fully-managed, highly available network load-balancing service. Load balancers distribute traffic to groups of droplets, which decouples the overall health of a backend service from the health of a single server to ensure that your services stay online.
```